jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererService.kt

```diff
@@ -463,9 +463,12 @@ class CodeWhispererService {
         // If contentSpans in reference are not consistent with content(recommendations),
         // remove the incorrect references.
         val validatedRecommendations = response.recommendations().map {
-            val validReference = it.hasReferences() && it.references().isNotEmpty() &&
-                it.content().length == it.references().last().recommendationContentSpan().end()
-            if (validReference) {
+            val validReferences = it.hasReferences() && it.references().isNotEmpty() &&
+                !it.references().any { reference ->
```

Comment:
@andrewyuq maybe we can change to use `it.references().none { reference -> ...}` instead of `!any()`